src/subcommand/init.rs

```diff
@@ -0,0 +1,104 @@
+#[tracing::instrument(skip_all)]
+pub fn main(data_root: &PathBuf) -> Result<(), Error> {
+    let default_already_exists = write_config(data_root, "config.yml", false)?;
+    write_config(data_root, "config.example.yml", true)?;
+    if default_already_exists {
+        let output = formatdoc! {r#"
+
+        `broker init` detected a previously existing config file at {config} and left it as is.
+
+        `broker init` did, however, create a new example config file for you at {example_config}.
+
+        This example config file contains a detailed explanation of everything you need to do to get broker up and running.
+
+        You can safely re-run `broker init` at any time to re-generate the "config.example.yml" file.
+        "#,
+            config = data_root.join("config.yml").display(),
+            example_config = data_root.join("config.example.yml").display(),
+        };
+        println!("{}", output);
+    } else {
+        let output = formatdoc! {r#"
+
+        `broker init` created an example config in {config}.
+
+        The config file contains a detailed explanation of everything you need to do to get broker up and running.
+
+        The next step is to open {config} and follow the instructions to configure broker.
+
+        We also wrote the same example file to {example_config} to serve as a reference for you in the future.
+
+        You can safely re-run `broker init` at any time to re-generate the "config.example.yml" file.
+
+        "#,
+            config = data_root.join("config.yml").display(),
+            example_config = data_root.join("config.example.yml").display(),
+        };
+        println!("{}", output);
+    }
+    Ok(())
+}
+
+fn write_config(data_root: &PathBuf, filename: &str, force_write: bool) -> Result<bool, Error> {
+    let config_file_path = data_root.join(filename);
+    if config_file_path.try_exists().unwrap_or(false) && !force_write {
+        return true.wrap_ok();
+    }
+
+    std::fs::create_dir_all(data_root)
+        .context_lazy(|| Error::CreateDataRoot(data_root.clone()))
+        .help_lazy(|| {
+            formatdoc! {r#"
+        `broker init` encountered an error while attempting to create the config directory {}.
+        This can happen if you do not have permission to create the directory.
+        Please ensure that you can create a directory at this location and try again
+        "#, data_root.display()}
+        })?;
+
+    fs::write(&config_file_path, default_config_file(data_root.as_path()))
+        .context_lazy(|| Error::WriteConfigFile(config_file_path.clone()))
+        .help_lazy(|| {
+            formatdoc! {r#"
+        `broker init` encountered an error while attempting to write a sample config file to {}.
+        This can happen if the you do not have permission to create files in that directory.
+        Please ensure that you can create a file at this location and try again
+        "#, config_file_path.display()}
+        })?;
```

Comment:
Same deal as above:

```
/// Errors encountered during init.
#[derive(Debug, thiserror::Error)]
pub enum Error {
    /// Writing the file did not work
    #[error("write config file inside data root '{}', at '{}'", .data_root.display(), .path.display())]
    WriteConfigFile {
        path: PathBuf,
        data_root: PathBuf,
    }
}

fs::write(&config_file_path, default_config_file(data_root))
    .context_lazy(|| Error::WriteConfigFile { 
        path: config_file_path.to_path_buf(),
        data_root: data_root.to_path_buf(),
    })
    .help_lazy(|| {
        formatdoc! {"
        This can happen if Broker did not have permission to create files in the data root directory.
        Ensure that your current user in the operating system is allowed to create files in the data root;
        deleting it and re-creating it may resolve this issue.
        Alternately, you may specify a different data root: run Broker with the `-h` argument to see how.
        "}
    })?;
```